lib/navigation/control/navigator.cpp

```diff
@@ -127,4 +132,65 @@ core::Result Navigator::accelerometerUpdate(
   return core::Result::kSuccess;
 }
 
-}  // namespace hyped::navigation
+void Navigator::publishCurrentTrajectory()
+{
+  const auto topic     = core::MqttTopic::kTest;
+  auto message_payload = std::make_shared<rapidjson::Document>();
+  message_payload->SetObject();
+
+  auto trajectory = currentTrajectory();
+  if (!trajectory) { logger_.log(core::LogLevel::kFatal, "Failed to get current trajectory"); }
+
+  rapidjson::Value displacement(trajectory->displacement);
+  rapidjson::Value velocity(trajectory->velocity);
+  message_payload->AddMember("displacement", displacement, message_payload->GetAllocator());
+  message_payload->AddMember("velocity", velocity, message_payload->GetAllocator());
+
+  const core::MqttMessage::Header header{.timestamp = 0,
+                                         .priority  = core::MqttMessagePriority::kNormal};
+  const core::MqttMessage message{topic, header, message_payload};
+  mqtt_->publish(message, core::MqttMessageQos::kExactlyOnce);
+}
+
+void Navigator::run()
+{
+  mqtt_->subscribe(core::MqttTopic::kKeyence);
+  mqtt_->subscribe(core::MqttTopic::kOpticalFlow);
+  mqtt_->subscribe(core::MqttTopic::kAccelerometer);
+
```

Comment:
can you make a `kStarted` topic and send a message on it saying that the navigation module has started successfully, we need to add something to deal with that stuff somewhere